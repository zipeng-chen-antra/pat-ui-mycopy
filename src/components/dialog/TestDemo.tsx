import React, { useState } from 'react';
import CSS from 'csstype';
import Dialog from './Dialog';
import DialogTitle from './DialogTitle';
import DialogContent from './DialogContent';
import DialogContentText from './DialogContentText';
import DialogAction from './DialogAction';

const buttonstyles: CSS.Properties= {
  display: "inline-flex",
  alignItems: "center",
  justifyContent: "center",
  position: "relative",
  boxSizing: "border-box",
  backgroundColor: "transparent",
  fontFamily: "Roboto Helvetica Arial sans-serif",
  fontWeight: 500,
  fontSize: "0.875rem",
  lineHeight: "1.75",
  textTransform: "uppercase",
  minWidth: "64px",
  padding: "5px 15px",
  borderRadius: "4px",
  border: "1px solid rgba(25, 118, 210, 0.5)",
  color: "#1976d2"
}

interface TestDemoProps {
  title?: string;
  content?: string;
  onClose?: () => void;
}

const TestDemo = ({title="Use Google's location service?", content="Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.", onClose = () => {}}: TestDemoProps): JSX.Element => {
  const [show, setShow] = useState<boolean>(false);
  const handleClose: () => void = () => {
    onClose();
    setShow(false);
  } 

  const handleClickButton: (e:React.MouseEvent<HTMLElement>) => void = (e) => {
    setShow(true);
  }

  return (
    <div>
      <button style={buttonstyles}
      onClick={handleClickButton}>click to open!</button>
      <Dialog show={show} onClose={handleClose}>
        <DialogTitle>{title}</DialogTitle>
        <DialogContent>
          <DialogContentText>{content}
          </DialogContentText>
        </DialogContent>
        <DialogAction>
          <button data-testid="cancel-button" onClick={handleClose}>cancel</button>
          <button onClick={handleClose}>Subscribe</button>
        </DialogAction>
      </Dialog>
    </div>)
};


export default TestDemo;