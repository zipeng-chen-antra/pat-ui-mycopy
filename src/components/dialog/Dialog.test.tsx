import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import TestDemo from './TestDemo';

describe('dialog', () => {
  it('should match snapshot', () => {
    const {asFragment} = render(<TestDemo/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should dialog pop out after user click the button', async () => {
    const {queryByTestId} = render(<TestDemo/>);
    expect(queryByTestId("dim-back")).toBeNull();
    let buttonEl = await screen.findByRole("button");
    fireEvent.click(buttonEl);
    expect(queryByTestId("dim-back")).toBeTruthy();
  });

  it('should dialog disappear after user click the back ground', async () => {
    const {queryByTestId} = render(<TestDemo/>);
    let buttonEl = await screen.findByRole("button");
    fireEvent.click(buttonEl);
    let dimBack = screen.getByTestId("dim-back");
    fireEvent.click(dimBack);
    expect(queryByTestId("dim-back")).toBeNull();
  });

  it('should fire the onClose function when the window is closed from the action button', async () => {
    const mockCallback = jest.fn();
    const {queryByTestId} = render(<TestDemo onClose={mockCallback}/>);
    let buttonEl = await screen.findByRole("button");
    fireEvent.click(buttonEl);
    let cancelButton = screen.getByTestId("cancel-button");
    fireEvent.click(cancelButton);
    expect(mockCallback).toBeCalledTimes(1);
  });

  it('should the window stop rolling when the dialog is pop out', async () => {
    const {queryByTestId} = render(<TestDemo/>);
    let buttonEl = await screen.findByRole("button");
    fireEvent.click(buttonEl);
    // let body = await screen.findBy("body");
    expect(document.body.style).toHaveProperty("overflow", "hidden");
  });
}

)