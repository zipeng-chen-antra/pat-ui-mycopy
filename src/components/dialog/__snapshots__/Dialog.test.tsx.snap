// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      style="display: inline-flex; align-items: center; justify-content: center; position: relative; box-sizing: border-box; background-color: transparent; font-family: Roboto Helvetica Arial sans-serif; font-weight: 500; font-size: 0.875rem; line-height: 1.75; text-transform: uppercase; min-width: 64px; padding: 5px 15px; border-radius: 4px; color: rgb(25, 118, 210);"
    >
      click to open!
    </button>
  </div>
</DocumentFragment>
`;
